src/crypto/OlmDevice.js

```diff
@@ -112,6 +105,32 @@ function OlmDevice(sessionStore) {
     this._inboundGroupSessionMessageIndexes = {};
 }
 
+/**
+ * Initialise the OlmAccount. This must be called before any other operations
+ * on the OlmDevice.
+ *
+ * Attempts to load the OlmAccount from localStorage, or creates one if none is
+ * found.
+ *
+ * Reads the device keys from the OlmAccount object.
+ */
+OlmDevice.prototype.init = async function() {
+    let e2eKeys;
+    const account = new Olm.Account();
+    try {
+        _initialise_account(this._sessionStore, this._pickleKey, account);
+        e2eKeys = JSON.parse(account.identity_keys());
+
+        this._maxOneTimeKeys = account.max_number_of_one_time_keys();
+    } finally {
+        account.free();
+    }
+
+    this.deviceCurve25519Key = e2eKeys.curve25519;
```

Comment:
This will throw if `_initialise_account` throws or if there are no identity keys as `e2eKeys` will be `undefined`.